src/runtime_src/core/pcie/driver/linux/xocl/xocl_xclbin.c

```diff
@@ -32,6 +32,12 @@ static int versal_xclbin_pre_download(xdev_handle_t xdev, void *args)
 	int ret = 0;
 
 	ret = xrt_xclbin_get_section(xclbin, PARTITION_METADATA, &metadata, &size);
+
+	/* aie only xclbin for edge_pcie platforms will not have partition metadata
+	   we should not error out by checking partition metadata */
+	if (xclbin->m_header.m_actionMask & AM_LOAD_AIE)
+		ret = 0;
```

Comment:
This is a bit confusing. Can we just move this check ahead and if Action Mask is set, just return. 
In AIE only XCLBIN, there will not be PARTITION_METADATA anyway. And we don't expect to tear down any URP sub driver either.